### PR TITLE
Update clock speed

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary

Update clock system, do this caused we want speed up the time calculation, so change:
 clock_time2ticks, clock_ticks2time, clock_timespec_add, clock_timespec_compare, clock_timespec_subtract
to MACRO

mainly include:

    clock: take clock_timespec_compare/add/subtract() as MACRO

    clock: Replace all ts and tick conversion functions
    
    Using the ts/tick conversion functions provided in clock.h

## Impact

clock

## Testing

Vela bes board, sim

### CODE REVIEW

**Comment by acassis** [comment]
@GUIDINGLI please fix typo: MARCO -> MACRO

Please add commit log explaining why to convert it to MACRO

**Comment by GUIDINGLI** [comment]
> @GUIDINGLI please fix typo: MARCO -> MACRO
> 
> Please add commit log explaining why to convert it to MACRO

done

**Comment by acassis** [comment]
> > @GUIDINGLI please fix typo: MARCO -> MACRO
> > Please add commit log explaining why to convert it to MACRO
> 
> done

Thank you @GUIDINGLI you just missed the commit message, please just explain there why to convert the function to MACRO

**Review by xiaoxiang781216 — APPROVED** [review]

File: sched/clock/clock_systime_ticks.c [hunk 1]
```diff
@@ -76,86 +76,51 @@
...
+   *    to the LS 32-bits for the first sample.
+   */
+
+  do
+    {
+      verify = g_system_ticks;
+      sample = g_system_ticks;
     }
+  while ((sample &  TIMER_MASK32)  < (verify &  TIMER_MASK32) ||
+         (sample & ~TIMER_MASK32) != (verify & ~TIMER_MASK32));
```
Comment (anchao) [inline sched/clock/clock_systime_ticks.c:120]: could we simplify the process here on architectures that support 64-bit atomicity

**Review by anchao — CHANGES_REQUESTED** [review]

File: sched/clock/clock_systime_ticks.c (hunk 1 above)
Comment (GUIDINGLI) [reply sched/clock/clock_systime_ticks.c:120]: Next patch

File: include/nuttx/clock.h [hunk 2]
```diff
@@ -722,6 +779,29 @@ void perf_convert(clock_t elapsed, FAR struct timespec *ts);
...
+
+/****************************************************************************
+ * Name: nxclock_gettime
+ *
+ * Description:
+ *   Get the current value of the specified time clock.
+ *
+ ****************************************************************************/
+
+void nxclock_gettime(clockid_t clock_id, struct timespec *tp);
```
Comment (anchao) [inline include/nuttx/clock.h:803]:
```suggestion
void nxclock_gettime(clockid_t clock_id, FAR struct timespec *tp);
```

**Pull request merged**